api/src/test/java/org/openmrs/module/fhir2/api/translators/impl/AllergyIntoleranceReactionComponentTranslatorImplTest.java

```diff
@@ -79,26 +77,58 @@ public void setUp() {
 	public void toFhirResource_shouldTranslateReactionToManifestation() {
 		Concept concept = new Concept();
 		concept.setUuid(CONCEPT_UUID);
+		String conceptName = "Test Reaction";
+		concept.addName(new ConceptName(conceptName, null));
 		
 		AllergyReaction reaction = new AllergyReaction();
 		reaction.setUuid(ALLERGY_REACTION_UUID);
 		reaction.setReaction(concept);
 		reaction.setAllergy(omrsAllergy);
-		reaction.setReactionNonCoded(NON_CODED_REACTION);
 		omrsAllergy.setReactions(Collections.singletonList(reaction));
 		
 		CodeableConcept codeableConcept = new CodeableConcept();
 		codeableConcept.addCoding(new Coding().setCode(CONCEPT_UUID));
-		codeableConcept.setText(NON_CODED_REACTION);
+		codeableConcept.setText(conceptName);
 		when(conceptTranslator.toFhirResource(concept)).thenReturn(codeableConcept);
+		
+		AllergyIntolerance.AllergyIntoleranceReactionComponent reactionComponent = reactionComponentTranslator
+		        .toFhirResource(omrsAllergy);
+		
+		assertThat(reactionComponent, notNullValue());
+		assertThat(reactionComponent.getManifestation().size(), equalTo(1));
+		assertThat(reactionComponent.getManifestation().get(0).getCoding().size(), equalTo(1));
+		assertThat(reactionComponent.getManifestation().get(0).getCoding().get(0).getCode(), equalTo(CONCEPT_UUID));
+		assertThat(reactionComponent.getManifestation().get(0).getText(), equalTo(conceptName));
+	}
+	
+	@Test
+	public void toFhirResource_shouldTranslateNonCodedReactionToManifestation() {
+		String nonCodedReaction = "Custom Reaction";
+		String otherConceptName = "Other";
+		
+		Concept otherConcept = new Concept();
+		otherConcept.setUuid(GLOBAL_PROPERTY_OTHER_VALUE);
+		otherConcept.addName(new ConceptName(otherConceptName, null));
+		
+		AllergyReaction reaction = new AllergyReaction();
+		reaction.setUuid(ALLERGY_REACTION_UUID);
+		reaction.setReaction(otherConcept);
+		reaction.setAllergy(omrsAllergy);
+		reaction.setReactionNonCoded(nonCodedReaction);
+		omrsAllergy.setReactions(Collections.singletonList(reaction));
+		
+		CodeableConcept codeableConcept = new CodeableConcept();
+		codeableConcept.addCoding(new Coding().setCode(CONCEPT_UUID));
+		when(conceptTranslator.toFhirResource(otherConcept)).thenReturn(codeableConcept);
+		
 		AllergyIntolerance.AllergyIntoleranceReactionComponent reactionComponent = reactionComponentTranslator
 		        .toFhirResource(omrsAllergy);
+		
 		assertThat(reactionComponent, notNullValue());
-		assertThat(reactionComponent.getManifestation(), hasSize(greaterThanOrEqualTo(1)));
-		assertThat(reactionComponent.getManifestation().get(0).getCoding(), hasSize(greaterThanOrEqualTo(1)));
-		assertThat(reactionComponent.getManifestation().size(), greaterThanOrEqualTo(1));
+		assertThat(reactionComponent.getManifestation().size(), equalTo(1));
+		assertThat(reactionComponent.getManifestation().get(0).getCoding().size(), equalTo(1));
```

Comment:
This results in better error messages, e.g., "Expected List of size 4, got list of size 3" instead of "Expected 4, got 3"

```suggestion
		assertThat(reactionComponent.getManifestation(), hasSize(equalTo(1)));
		assertThat(reactionComponent.getManifestation().get(0).getCoding(), hasSize(equalTo(1)));
```